### PR TITLE
DNM [nrf mergeup] Upmerge 180121

### PR DESCRIPTION
**Pull request by jfischer-no**
Synchronize Zephyr up to upstream commit
ff720cd9b343bd1c2764e0aca7ae91f31832581b

### CODE REVIEW

File: subsys/dfu/img_util/flash_img.c [hunk 1]
```diff
@@ -18,6 +18,7 @@
 #endif
 
 #include <devicetree.h>
+
```
Comment (nvlsianpu) [inline subsys/dfu/img_util/flash_img.c]: remove this empty line from reapplied `[nrf noup] tree-wide: support NCS Partition Manager (PM) definitions.`

**Review by rlubos — APPROVED** [review]

**Review by mbolivar-nordic — APPROVED** [review]
LGTM. Here is a linear history with no diff from this PR as of 21e51edb940e20566d57764d62960e08e85ac852, just for reference:

https://github.com/mbolivar-nordic/fw-nrfconnect-zephyr/tree/upmerge-427-rebase

@jfischer-no can you please let me know if you do anything to the code besides adding new commits on top of 21e51edb940e20566d57764d62960e08e85ac852 or rewording existing commit messages in this PR?

Constructing that linear history took longer than usual because I had to manually work around an issue in our existing master branch history, so if you change the structure of the PR, I'd like to know so I can do this work again in time to prepare for the "real" history rewrite with correct `cherry-pick -x` output.

**Review by trond-snekvik — APPROVED** [review]

**Review by joerchan — APPROVED** [review]

**Comment by jfischer-no** [comment]
> @jfischer-no can you please let me know if you do anything to the code besides adding new commits on top of [21e51ed](https://github.com/nrfconnect/sdk-zephyr/commit/21e51edb940e20566d57764d62960e08e85ac852) or rewording existing commit messages in this PR?

@mbolivar-nordic I see that I forgot to remove unintentional change in [subsys/dfu/img_util/flash_img.c](https://github.com/nrfconnect/sdk-zephyr/pull/427/files/8370f194d2d2f96eebf4871eb4220eff056e4218..7bef08cb78bd97abf69149858dfce7094c4d24a8#diff-337a69db622d3a4c11b762a627c9b5fedba6029b2cd17dde5433214cc9f34a27). Is it okay for you if I fix "_[nrf noup] tree-wide: support NCS Partition Manager (PM) definitions_"  accordingly?

**Comment by edmont** [comment]
If possible, I would like to include these 3 commits into this upmerge: https://github.com/zephyrproject-rtos/zephyr/pull/31598/commits

They provide a couple of bugfixes for OpenThread (KRKNWK-8770, KRKNWK-8771).

If not possible, I'll cherry-pick them after the upmerge.

**Comment by jfischer-no** [comment]
> If possible, I would like to include these 3 commits into this upmerge: https://github.com/zephyrproject-rtos/zephyr/pull/31598/commits
> 
> They provide a couple of bugfixes for OpenThread (KRKNWK-8770, KRKNWK-8771).
> 
> If not possible, I'll cherry-pick them after the upmerge.

@edmont Are there any incompatibilities with sdk-nrf?

**Comment by edmont** [comment]
> @edmont Are there any incompatibilities with sdk-nrf?

There won't be build issues but maybe some Thread tests can fail, there is a pending task to fix those after the OpenThread changes: KRKNWK-8507.

**Comment by jfischer-no** [comment]
> > @edmont Are there any incompatibilities with sdk-nrf?
> 
> There won't be build issues but maybe some Thread tests can fail, there is a pending task to fix those after the OpenThread changes: KRKNWK-8507.

Then let's do it after upmerge.

**Comment by edmont** [comment]
> Then let's do it after upmerge.

Sure, no problem.

**Comment by nvlsianpu** [comment]
@jfischer-no Herere fix for inherited issue https://github.com/zephyrproject-rtos/zephyr/issues/31629 https://github.com/jfischer-no/sdk-zephyr/pull/2 (https://github.com/zephyrproject-rtos/zephyr/pull/31626)

File: include/net/socket.h [hunk 2]
```diff
@@ -821,6 +822,12 @@ static inline char *inet_ntop(sa_family_t family, const void *src, char *dst,
 /** sockopt: disable IPv6 ICMP replies */
 #define SO_IPV6_ECHO_REPLY 32
 
+/**
+ * sockopt: Receive timeout
+ * Applies to receive functions like recv(), but not to connect()
+ */
+#define SO_RCVTIMEO 20
```
Comment (utzig) [inline include/net/socket.h:829]: This "macro" is already defined a few lines above which is what is causing the documentation build issue.

Comment (jfischer-no) [reply include/net/socket.h:829]: I see, the first one is from "[nrf noup] include: net: add NCS extensions" the second from upstream. Thanks @utzig

Comment (jfischer-no) [reply include/net/socket.h:829]: That would require one revert further than before, so I went one more to the last [nrf temphack].

**Comment by jfischer-no** [comment]
@mbolivar-nordic, @hakonfam, @de-nordic there are three [nrf temphack] from previous upmerge, please check if they are still necessary / up-to-date.

**Review by nvlsianpu — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Review by de-nordic — APPROVED** [review]

File: subsys/storage/flash_map/flash_map_default.c [hunk 3]
```diff
@@ -10,10 +10,48 @@
 #include <zephyr.h>
 #include <storage/flash_map.h>
 
-#define FLASH_AREA_FOO(part)						\
```
Comment (nvlsianpu) [inline subsys/storage/flash_map/flash_map_default.c]: e571c885653830bc643e27f1e920018a1241d6e6 reworks what goes from zephyr vanilla, this need to be reflected in
`[nrf noup] tree-wide: support NCS Partition Manager (PM) definitions` commit

**Review by nvlsianpu — CHANGES_REQUESTED** [review]

File: subsys/storage/flash_map/flash_map_default.c (hunk 3 above)
Comment (nvlsianpu) [reply subsys/storage/flash_map/flash_map_default.c]: The NCS commit should just introduce the alternative inside for `#if USE_PARTITION_MANAGER` case

**Comment by nvlsianpu** [comment]
`[nrf noup] drivers: flash: No spm_request_read when no secure services 83a613399c4780c8098ff521566fc48d57c5dc47` can be melded into `[nrf noup] tree-wide: support NCS Partition Manager (PM) definitions`

File: subsys/storage/flash_map/flash_map_default.c (hunk 3 above)
Comment (jfischer-no) [reply subsys/storage/flash_map/flash_map_default.c]: fixed, please recheck

**Review by nvlsianpu — APPROVED** [review]

**Review by tejlmand — APPROVED** [review]

**Pull request merged**